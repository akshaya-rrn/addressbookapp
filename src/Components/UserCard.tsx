import { CardActionArea, CardContent, CardMedia, Grid, Typography } from '@mui/material';
import Card from '@mui/material/Card';
import React from 'react';
import { Link } from 'react-router-dom';
import { UserResponseResult } from '../types';
import { getUserName } from '../utility/Utils';

interface UsersCardComponentProps {
  usersData: UserResponseResult[];
}

interface DisplayCardProps {
    userInfo: UserResponseResult;
}

export const UsersCard: React.FC<UsersCardComponentProps> = ({usersData}: UsersCardComponentProps) => {
    
    const DisplayUserCard: React.FC<DisplayCardProps> = ({userInfo} : DisplayCardProps) => {
        const { name, picture, email } = userInfo;
        
        return (
            <Card sx={{ maxWidth: 300 }}>
            <CardMedia component="img" height="140" src={picture.medium} />
            <CardContent>
            <Typography gutterBottom variant="h6" component="div">{getUserName(name)}</Typography>
            <Typography variant="body2" color="text.secondary">{email}<br/>
            </Typography>
            <CardActionArea>
              <br/>
                <Link to="/detailedview" state={{user: userInfo}}> Click for Details </Link>
            </CardActionArea>
            </CardContent>
        </Card>  
        );
  };

  return (
    <Grid id="useraddressbook" container spacing={5}>
      {usersData.map((user, index) => (
        <Grid key={index} item xs={2}>
          <DisplayUserCard userInfo={user} />
        </Grid>
      ))}
    </Grid>
  );
};
