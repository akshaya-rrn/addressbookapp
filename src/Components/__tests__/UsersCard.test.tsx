import renderer from 'react-test-renderer';

import { BrowserRouter as Router } from 'react-router-dom';
import { UsersCard } from '../UserCard';
import { mockgetUsersResponse } from './mockData/getUsersData';


describe('UsersCard Component', () => {

  it('Users Card component is rendered correctly', () => {
    var response = JSON.parse(JSON.stringify(mockgetUsersResponse));
    const tree = renderer.create(<Router><UsersCard usersData={response.results}/></Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

