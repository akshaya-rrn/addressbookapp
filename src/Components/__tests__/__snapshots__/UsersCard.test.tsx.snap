// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersCard Component Users Card component is rendered correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 css-1ld3b9g-MuiGrid-root"
  id="useraddressbook"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1xez8po-MuiPaper-root-MuiCard-root"
    >
      <img
        className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
        height="140"
        src="https://randomuser.me/api/portraits/med/men/41.jpg"
      />
      <div
        className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
        >
          Momčilo Danilović
        </div>
        <p
          className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          momcilo.danilovic@example.com
          <br />
        </p>
        <button
          className="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <br />
          <a
            href="/detailedview"
            onClick={[Function]}
          >
             Click for Details 
          </a>
          <span
            className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1xez8po-MuiPaper-root-MuiCard-root"
    >
      <img
        className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
        height="140"
        src="https://randomuser.me/api/portraits/med/men/69.jpg"
      />
      <div
        className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
        >
          Ermitério Vieira
        </div>
        <p
          className="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        >
          ermiterio.vieira@example.com
          <br />
        </p>
        <button
          className="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <br />
          <a
            href="/detailedview"
            onClick={[Function]}
          >
             Click for Details 
          </a>
          <span
            className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
